### DISCUSSION TITLE
Is this possible to Generate a Release Notes by comparing in between a list??

### DISCUSSION THREAD

**Discussion by purushothamb11** [rfennell/AzurePipelines#916] [Q&A] [answered]
Hello everyone, 
let's think we have 2 environments ( Dev and QA ), here Dev will be having releases regularly and every time a new release notes gets generated, but for QA the release is going to happen only twice a month or once in a month and it will generate only one or two release notes per month so is it possible to get all the commits & ids of all releases from Dev when releasing to QA.

Example : Dev has total 15 releases and it generated 15 release notes, but when deploying to QA at release 16 it  is generating only the associated commits & ids from release 15, so is there any possibility to get all the associated commits and id's from release 1 to 15 in release 16.

Please let me know and Thanks in advance...

**Comment by rfennell** [comment, marked as answer]
Yes I think you can, there are a number of ways. I would suggest using the release note task twice in the Dev Stage

1. As you do now generate the release notes per build
2. Also have the release note task set to generate the release not but comparing a fixed build run ID, or against the last build that was marked with a given tag.

The other option is to move the send generate release note to the QA stage, but I think the outline above is closer to what you requested

You can see all the parameter options in the [WIKI](https://marketplace.visualstudio.com/items?itemName=richardfennellBM.BM-VSTS-XplatGenerateReleaseNotes)
